programs/mpl-core/src/state/update_authority.rs

```diff
@@ -75,14 +75,17 @@ impl UpdateAuthority {
                     PluginType::UpdateDelegate,
                 );
 
-                if let Ok((authority, _, _)) = maybe_update_delegate {
+                if let Ok((authority, update_delegate_plugin, _)) = maybe_update_delegate {
                     if assert_collection_authority(&collection, authority_info, &authority).is_err()
                         && assert_collection_authority(
                             &collection,
                             authority_info,
                             &Authority::UpdateAuthority,
                         )
                         .is_err()
+                        && !update_delegate_plugin
+                            .additional_delegates
+                            .contains(authority_info.key)
```

Comment:
Why is the `update_delegate_plugin` handling in the implementation for `UpdateAuthority::validate_create` rather than the implementation for `UpdateDelegate::validate_create`?

Reply:
I'm not sure if it's just because Create is special but I would be open to refactoring that in a separate PR when we take a look at that whole logic flow.

Reply:
sg